src/test/kotlin/no/skatteetaten/aurora/boober/feature/S3StorageGridFeatureTest.kt

```diff
@@ -0,0 +1,138 @@
+import no.skatteetaten.aurora.boober.service.resourceprovisioning.ObjectAreaWithCredentials
+import no.skatteetaten.aurora.boober.service.resourceprovisioning.S3ObjectArea
+import no.skatteetaten.aurora.boober.service.resourceprovisioning.S3StorageGridProvisioner
+import no.skatteetaten.aurora.boober.service.resourceprovisioning.StorageGridCredentials
+import no.skatteetaten.aurora.boober.utils.AbstractFeatureTest
+import no.skatteetaten.aurora.boober.utils.findResourcesByType
+import no.skatteetaten.aurora.mockmvc.extensions.mockwebserver.HttpMock
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+
+class S3StorageGridFeatureTest : AbstractFeatureTest() {
+    val s3StorageGridProvisioner = mockk<S3StorageGridProvisioner>()
+
+    override val feature: Feature
+        get() {
+            return S3StorageGridFeature(
+                s3StorageGridProvisioner,
+                mockk(),
+            )
+        }
+
+    @AfterEach
+    fun after() {
+        HttpMock.clearAllHttpMocks()
+    }
+
+    val tenant = "paas-utv"
+    val bucketName = "theBucket"
+
+    @Test
+    fun `creates secretes and environment variable refs for provisioned credentials`() {
+
+        val area1Name = "default"
+        val area2Name = "min-andre-bucket"
+
+        every { s3StorageGridProvisioner.getOrProvisionCredentials(any()) } returns listOf(
+            ObjectAreaWithCredentials(
+                S3ObjectArea(tenant, bucketName, area1Name),
+                StorageGridCredentials(
+                    tenant,
+                    "endpoint",
+                    "accessKey",
+                    "secretKey",
+                    bucketName,
+                    "objectPrefix",
+                    "username",
+                    "password"
+                )
+            ),
+            ObjectAreaWithCredentials(
+                S3ObjectArea(tenant, bucketName, area2Name),
+                StorageGridCredentials(
+                    tenant,
+                    "endpoint",
+                    "accessKey",
+                    "secretKey",
+                    bucketName,
+                    "objectPrefix",
+                    "username",
+                    "password"
+                )
+            )
+        )
+
+        val resources = generateResources(
+            """{ 
+                "s3Defaults": {
+                    "bucketName": "theBucket",
+                    "tenant": "paas-utv"
+                },
+                "s3": {
+                    "$area1Name": {
+                        "enabled": true
+                    },
+                    "$area2Name": {
+                        "enabled": true
+                    }
+                }
+           }""",
+            resources = mutableSetOf(createEmptyApplicationDeployment(), createEmptyDeploymentConfig()),
+            createdResources = 2
+        )
+
+        resources.verifyS3SecretsAndEnvs(listOf(area1Name, area2Name))
```

Comment:
Burde vi heller sammenligne mot filer slik vi gjør for resten av feature tester?